src/librustc/mir/interpret/step.rs

```diff
@@ -286,12 +286,13 @@ impl<'a, 'b, 'tcx, M: Machine<'tcx>> Visitor<'tcx> for ConstantExtractor<'a, 'b,
             match constant.literal {
                 // already computed by rustc
                 mir::Literal::Value { value: &ty::Const { val: ConstVal::Unevaluated(def_id, substs), .. } } => {
-                    debug!("global_item: {:?}, {:?}", def_id, substs);
+                    debug!("global_item: {:?}, {:#?}", def_id, substs);
                     let substs = this.ecx.tcx.trans_apply_param_substs(this.instance.substs, &substs);
-                    let param_env = this.ecx.tcx.param_env(def_id);
+                    debug!("global_item_new_substs: {:#?}", substs);
+                    debug!("global_item_param_env: {:#?}", M::param_env(this.ecx));
                     let instance = Instance::resolve(
                         this.ecx.tcx,
-                        param_env,
+                        M::param_env(this.ecx),
```

Comment:
Why does the machine control the `param_env`? Is this only for `Reveal`?

Reply:
I added it at rustfest on @arielb1's comment that we need to track the param env. I have no idea what I'm doing there.

Reply:
Well I would've expected it to be always around as part of one of the contexts instead of being in the machine.

Reply:
Ah... well. The miri interpreter doesn't store it, it just uses reveal all